### PR TITLE
feat: /upgrade unit-testing

### PR DESCRIPTION
**Pull request by odyslam**
## Solution

- UpgradeBeacon.t.sol
- UpgradeBeaconController.t.sol
- UpgradeBeaconProxy.t.sol

## PR Checklist

- [ ] Added Tests
- [ ] Updated Documentation
- [ ] Updated CHANGELOG.md for the appropriate package

### CODE REVIEW

File: packages/contracts-core/contracts/test/utils/MockImpl.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+
+contract MockImpl {
+    bool public fires;
```
Comment (prestwich) [inline packages/contracts-core/contracts/test/utils/MockImpl.sol:5]: cute

**Review by prestwich — CHANGES_REQUESTED** [review]

File: packages/contracts-core/contracts/test/UpgradeBeacon.t.sol [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            abi.encodeWithSignature("Error(string)", "implementation !contract")
+        );
+    }
+
+    function test_fallbackControllerFailNotContractFuzzed(address newImpl)
+        public
+    {
+        vm.assume(!Address.isContract(newImpl));
+        // any address that is not a EOA
+        address newImpl = address(0xBEEFEEF);
```
Comment (prestwich) [inline packages/contracts-core/contracts/test/UpgradeBeacon.t.sol:87]: this line breaks the fuzz

Comment (odyslam) [reply packages/contracts-core/contracts/test/UpgradeBeacon.t.sol:87]: it passes on my machine, and it looks fine. Not sure what you mean

**Comment by prestwich** [comment]
youre having rhe fuzzer pass in newImpl, then immediately shadowing it with
a constant. no fuzzing is being done as a result

**Comment by odyslam** [comment]
@prestwich duh, fixing it

**Review by prestwich — APPROVED** [review]

**Pull request merged**